spec/predicates/runtime-trace.md

```diff
@@ -0,0 +1,2712 @@
+instance of the job being observed. The `monitorLog` field contains the actual
+runtime trace information.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v0.1",
+    "subject": [{ ... }],
+    "predicateType": "https://in-toto.io/attestation/runtime-trace/v0.1",
+    "predicate": {
+        "monitor": {
+            "id": "<URI>",
+            "configSource": "<ResourceDescriptor>",
+            "tracePolicy": { /* object */ }
+        },
+        "monitoredProcess": {
+            "hostID": "<URI>",
+            "type": "<URI>",
+            "event": "<STRING>"
+        },
+        "monitorLog": {
+            "process": [
+                { /* object */ }
+            ],
+            "network": [
+                { /* object */ }
+            ],
+            "fileAccess": [
+                { /* object */ }
+            ]
+        },
+        "metadata": {
+            "buildStartedOn": "<TIMESTAMP>",
+            "buildFinishedOn": "<TIMESTAMP>"
+        }
+    }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the in-toto attestation parsing rules. Summary:
+
+-   Consumers MUST ignore unrecognized fields.
+-   The `predicateType` URI includes the major version number and will always
+    change whenever there is a backwards incompatible change.
+-   Minor version changes are always backwards compatible and “monotonic.” Such
+    changes do not update the `predicateType`.
+-   Producers MAY add extension fields using field names that are URIs.
+-   Fields marked _optional_ MAY be unset or null, and should be treated
+    equivalently. Both are equivalent to empty for object or array values.
+
+## Fields
+
+`monitor` _object_, _required_
+
+Identifies the specific monitor instance used to trace the runtime.
+
+`monitor.id` _string (URI)_, _required_
+
+URI indicating the monitor’s identity.
+
+`monitor.configSource` ResourceDescriptor, _optional_
+
+Effectively a pointer to the monitor's configuration.
+
+`monitor.tracePolicy` _object_, _optional_
+
+Indicates the trace policy used for the monitoring event that generated the
+runtime trace. The trace policy's format is dependent on the monitor used, and
+so it must be parsed after identifying the monitor using `monitor.id`.
+
+FIXME: should this be optional?
+
+`monitoredProcess` _object_, _required_
+
+Identifies the process being monitored.
+
+`monitoredProcess.hostID` _string (URI)_, _required_
+
+URI indicating the process host’s identity. Ex: GitLab Runner ID
+
+`monitoredProcess.type` _string (TypeURI)_, _required_
+
+URI indicating the type of process performed. Ex: when monitoring a build, this
+field records the build type.
+
+`monitoredProcess.event` _string_, _required_
+
+String identifying the specific job or task associated with the attestation.
+
+`monitorLog` _object_, _required_
+
+Record of events that were traced by the monitor. One of `process`, `network`,
+and `fileAccess` must be present.
+
+`monitorLog.process` _list_, _optional_
```

Comment:
I think we should include them and indicate that their exact format is still being worked upon by cross referencing different monitor tools? Until then, it may be determined using the monitor ID? cc @pxp928

Reply:
I've updated the text to reflect this, happy to change it further depending on what everyone thinks.